### PR TITLE
[DOCFIX] Add overview of stress bench

### PR DESCRIPTION
**Pull request by HelloHorizon**
### What changes are proposed in this pull request?

Add overview document for StressBench framework. This pr is only for the overview of the StressBench, will add the details for each benchmarks tool afterward.

### Why are the changes needed?

N/A

### Does this PR introduce any user facing changes?

N/A

### CODE REVIEW

File: docs/en/operation/StressBench.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+StressBench test suite is a performance benchmark suite to help you understand the performance of
+Alluxio in different scenarios. It will help you quickly test and analyze the Alluxio usage in your
+environment by simply running the command line.
+
+The StressBench is consisted of different tests to help you test the metadata and IO performance of
+the Alluxio system. The following benchmark tools are currently supported:
+* `StressMasterBench` - A benchmark tool measuring the master performance of Alluxio.
+* `StressClientBench` - A benchmark tool measuring the IO performance of Alluxio through the client.
+* `StressWorkerBench` - A benchmark tool measuring the IO performance of Alluxio with single node.
+* `UfsIOBench` - A benchmark tool measuring the IO performance to UFS.
```
Comment (HelloHorizon) [inline docs/en/operation/StressBench.md:22]: @jiacheliu3 is this a fair statement for UfsIOBench?

File: docs/en/operation/StressBench.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+* `FuseIOBench` - A benchmark tool measuring the IO performance of Alluxio through Fuse interface.
```
Comment (HelloHorizon) [inline docs/en/operation/StressBench.md:23]: @LuQQiu @ssz1997 is this a fair statement for `FuseIOBench`?

File: docs/en/operation/StressBench.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
```
Comment (HelloHorizon) [inline docs/en/operation/StressBench.md:24]: @jiacheliu3 is there any one-line summary for `RPCBench`?

File: docs/en/operation/StressBench.md (hunk 2 above)
Comment (LuQQiu) [reply docs/en/operation/StressBench.md:23]: Yeah, it looks good

**Review by LuQQiu — APPROVED** [review]
LGTM, thanks for adding the doc!

File: docs/en/operation/StressBench.md [hunk 4]
```diff
@@ -36,10 +36,9 @@ further coordinate distributed tasks for the job.
 ### Distributed Tasks
 These tasks are run on the job worker, and eventually report back to the job master.
 
-## More Consistent Results
-When testing results are inconsistent and wildy variable, it makes trusting the
-results more difficult. Therefore, there are some additional features that help
-StressBench to be more consistent with results.
+## Consistent Results
+When testing results are inconsistent and wildy variable, it makes trusting the results more
```
Comment (jiacheliu3) [inline docs/en/operation/StressBench.md:40]:
```suggestion
When testing results are inconsistent and wildly variable, it makes trusting the results more
```

**Review by jiacheliu3 — CHANGES_REQUESTED** [review]
Found a typo

File: docs/en/operation/StressBench.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+* Table of Contents
+  {:toc}
+
+The Alluxio system runs in different environments with different hardware for various purposes. The
+StressBench test suite is a performance benchmark suite to help you understand the performance of
+Alluxio in different scenarios. It will help you quickly test and analyze the Alluxio usage in your
+environment by simply running the command line.
+
+The StressBench is consisted of different tests to help you test the metadata and IO performance of
+the Alluxio system. The following benchmark tools are currently supported:
```
Comment (jiacheliu3) [inline docs/en/operation/StressBench.md:18]:
```suggestion
The StressBench is consisted of different tests to help you measure various aspects of the Alluxio system including 
metadata and IO. The following benchmark tools are currently supported:
```

File: docs/en/operation/StressBench.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+* `StressMasterBench` - A benchmark tool measuring the master performance of Alluxio.
+* `StressClientBench` - A benchmark tool measuring the IO performance of Alluxio through the client.
+* `StressWorkerBench` - A benchmark tool measuring the IO performance of Alluxio with single node.
+* `UfsIOBench` - A benchmark tool measuring the IO performance to UFS.
```
Comment (jiacheliu3) [inline docs/en/operation/StressBench.md:22]:
```suggestion
* `UfsIOBench` - A benchmark tool measuring the IO throughput between the Alluxio cluster and the UFS.
```

File: docs/en/operation/StressBench.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+The Alluxio system runs in different environments with different hardware for various purposes. The
+StressBench test suite is a performance benchmark suite to help you understand the performance of
+Alluxio in different scenarios. It will help you quickly test and analyze the Alluxio usage in your
+environment by simply running the command line.
+
+The StressBench is consisted of different tests to help you test the metadata and IO performance of
+the Alluxio system. The following benchmark tools are currently supported:
+* `StressMasterBench` - A benchmark tool measuring the master performance of Alluxio.
+* `StressClientBench` - A benchmark tool measuring the IO performance of Alluxio through the client.
+* `StressWorkerBench` - A benchmark tool measuring the IO performance of Alluxio with single node.
```
Comment (jiacheliu3) [inline docs/en/operation/StressBench.md:21]: I feel the description are a little unclear. For example for the "master performance of Alluxio" maybe it's better to mention the metadata operations? Same for the other two commands. Maybe we can further clarify what "through the client" means to a user? Also is `StressWorkerBench` using only a single node (I don't know how this test works but this descriptions makes me want to ask)?

File: docs/en/operation/StressBench.md (hunk 1 above)
Comment (jiacheliu3) [reply docs/en/operation/StressBench.md:22]: See suggested change

File: docs/en/operation/StressBench.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+### Distributed Tasks
+These tasks are run on the job worker, and eventually report back to the job master.
+
+## Consistent Results
+When testing results are inconsistent and wildy variable, it makes trusting the results more
+difficult. Therefore, there are some features that help StressBench to be consistent with results.
+
+### Barriers
+Barriers are one way that StressBench tries to achieve consistent results. This allows distributed
```
Comment (jiacheliu3) [inline docs/en/operation/StressBench.md:44]: How are the start time determined? I feel like using a start time heuristically does not give something equivalent to what we mean typically by "Barrier" with signals and conditions? (I'm not sure how to improve the description here).

File: docs/en/operation/StressBench.md (hunk 3 above)
Comment (jiacheliu3) [reply docs/en/operation/StressBench.md:24]: Actually `RpcBench` is a parent class with 3 children implementations. See if the lines below makes sense

```
RpcBench - A benchmark tool which simulates RPCs with a specified load and concurrency. Below are currently 3 implementations of RpcBench, each simulating one specific RPC. This tool is used to incur RPC stress on the master in a controlled manner.
RegisterWorkerBench - An implementation of RpcBench which simulates the RegisterWorker RPC.
WorkerHeartbeatBench - An implementation of RpcBench which simulates the WorkerHeartbeat RPC.
GetPinnedFileIdsBench - An implementation of RpcBench which simulates the GetPinnedFileIds RPC.
```

File: docs/en/operation/StressBench.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+of the job workers to be in sync. Then, the tasks themselves can use that start time to ensure all
+tasks and threads start at the defined time. This ensures that all distributed tasks and threads
+start at the same time for the testing.
+
+### Duration-based testing (with warmups)
+Another way to improve the consistency of results is to run tests with warmups and also for a
+defined amount of time. If a particular thread or task is finished first, it would also provide any
+load to the system instead of ending with an idle stage. That means tasks could start at the same
+times and end at the same times, which will ensure the system is under load during the whole testing
+time.
```
Comment (jiacheliu3) [inline docs/en/operation/StressBench.md:56]:
```suggestion
period.
```

**Review by jiacheliu3 — CHANGES_REQUESTED** [review]
Added more comments specific to each section. Thanks for adding this doc! It fills a lot of gaps.

Taking a step back, I feel like we can add more sections/info to this doc page to provide more details:
1. I'm not sure if we should break the commands into groups. For each group there are some common parameters that control behavior (like `--duration`). These parameters apply well to some tests but not to others. For example there are the stress testing family and the RPC simulator family, if Fuse does not make its own family (or maybe UfsIO belongs to that family too).
2. More examples for each test command will definitely help a lot. For example when we introduce `--duration` the users have no idea how to use them. We need to have some sections like https://docs.alluxio.io/os/user/stable/en/operation/User-CLI.html#runufsiotest explaining how the arguments should be fed. Most likely one example does not suffice for each command.

There's no end to doc improvements, so I won't let these comments block your PR. Having as much of these included as possible would be awesome, but we can always add it in subsequent PRs. What do you think?

File: docs/en/operation/StressBench.md (hunk 8 above)
Comment (HelloHorizon) [reply docs/en/operation/StressBench.md:46]: https://en.wikipedia.org/wiki/Barrier_(computer_science) barrier is a fair statement here.
You can check https://github.com/Alluxio/alluxio/blob/v2.4.1/stress/shell/src/main/java/alluxio/stress/cli/StressMasterBench.java#L224 to see how we define the timestamp of when the tasks should start in `StressMasterBench`
, and https://github.com/Alluxio/alluxio/blob/v2.4.1/job/server/src/main/java/alluxio/job/plan/stress/StressBenchDefinition.java#L161 for `start time`.

File: docs/en/operation/StressBench.md (hunk 3 above)
Comment (HelloHorizon) [reply docs/en/operation/StressBench.md:26]: I put the summary of `RpcBench` in this section, we can put the details of RPCBench in RPCBench section in following pr

**Review by HelloHorizon — COMMENTED** [review]
@jiacheliu3 Thanks for the comments.
We can add/modify the description in the following pr, in which we will give a detailed explanation of each StressBench tool. And since we still actively implement more stress bench tools, like fusebench and rpcbench, some statements here may not be true afterward. Let's keep this structure for now and modify them as needs in the following pr.

**Comment by HelloHorizon** [comment]
alluxio-bot, merge this please

**Pull request merged**